colabs/Optymalizacja_i_rozbudowa_similarity.ipynb

```diff
@@ -0,0 +1,850 @@
+              "    }\n",
+              "  </style>\n",
+              "\n",
+              "      <script>\n",
+              "        const buttonEl =\n",
+              "          document.querySelector('#df-540dedd0-4b3f-4911-a5c4-afb1909c431d button.colab-df-convert');\n",
+              "        buttonEl.style.display =\n",
+              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
+              "\n",
+              "        async function convertToInteractive(key) {\n",
+              "          const element = document.querySelector('#df-540dedd0-4b3f-4911-a5c4-afb1909c431d');\n",
+              "          const dataTable =\n",
+              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
+              "                                                     [key], {});\n",
+              "          if (!dataTable) return;\n",
+              "\n",
+              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
+              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
+              "            + ' to learn more about interactive tables.';\n",
+              "          element.innerHTML = '';\n",
+              "          dataTable['output_type'] = 'display_data';\n",
+              "          await google.colab.output.renderOutput(dataTable, element);\n",
+              "          const docLink = document.createElement('div');\n",
+              "          docLink.innerHTML = docLinkHtml;\n",
+              "          element.appendChild(docLink);\n",
+              "        }\n",
+              "      </script>\n",
+              "    </div>\n",
+              "  </div>\n",
+              "  "
+            ]
+          },
+          "metadata": {},
+          "execution_count": 11
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Przy liczeniu similarity dla obciętego do niezbędnych kolumn zbioru danych, ustalamy parametr `reduced_dataset` w metodzie `similarity_one_vs_all` na `True`."
+      ],
+      "metadata": {
+        "id": "U988JEtDFOX3"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "VFBGGA26Z-rF",
+        "outputId": "3c7e01a4-bff4-4f79-80ce-b465da1cd86a",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        }
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "CPU times: user 481 ms, sys: 70 ms, total: 551 ms\n",
+            "Wall time: 506 ms\n"
+          ]
+        }
+      ],
+      "source": [
+        "%%time\n",
+        "ids, values = similarity_one_vs_all(sample, Arnold_schwarzenegger, reduced_dataset=True)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "values[:3]"
```

Comment:
ja nie widzę osatecznie, kto jest najbardziej podobny do tego aktora

Reply:
Bo do tego jest funkcja ranking. Tutaj tylko wyświetlałem początek outputu tych funkcji. Jeżeli jest potrzeba to dodam przykład aplikacji rankingu

Reply:
Zostało dodany ranking 3 aktorów w postaci ich id oraz imion